### PR TITLE
Fix issue with allowPartialMatch

### PR DESCRIPTION
**Pull request by youming-lin**
## Description

Add more unit tests, add logic to correct behavior of `allowPartialMatch`.

## Motivation and Context

#821

## How Has This Been Tested?

More unit tests

## Checklist:

- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by youming-lin** [comment]
@shmuelk Can you review this?

File: Sources/Kitura/RouterElement.swift [hunk 1]
```diff
@@ -41,6 +41,9 @@ class RouterElement {
     /// The middlewares to use
     private let middlewares: [RouterMiddleware]
 
+    /// allowPartialMatch flag
```
Comment (ianpartridge) [inline Sources/Kitura/RouterElement.swift]: Useless comment :) Can we have a useful one?

Comment (youming-lin) [inline Sources/Kitura/RouterElement.swift]: Sure can :)

**Comment by youming-lin** [comment]
Rebased branch on top of latest `master`, added `mergeParameters` fix, and updated comments.

@shmuelk or @ianpartridge Can you review again?

**Comment by shmuelk** [comment]
Thanks for the changes as we discussed.

**Pull request merged**